### DISCUSSION TITLE
Experience System

### DISCUSSION THREAD

**Discussion by Indy2222** [DigitalExtinction/Game#80] [Ideas]
Experience in an activity makes units / buildings better in that activity. It grows exponentially, id est gain is multiplicative and not additive.

Experience is not unit specific, but global since "AI" can "upload" the experience.

### Combat

During combat, user gains experience. This makes all units more capable during (future) combat. The experience grows with any combat, but mostly when enemy targets are destroyed. Harder to destroy targets should have disproportionately higher experience impact.

Question: how does combat "capability" manifest?

### Construction

There is specific and general construction experience. Constructing a thing makes the player better in constructing anything but it has larger impact on that specific thing.

Question: how does construction "capability" manifest?

### Research

Research experience affects research progress per unit of time. Research in one area affects all other research areas.

**Comment by JackCrumpLeys** [comment]
> Question: how does combat "capability" manifest?

one idea is to specify variables in the object definition that have a certain 'step' value as to how much it increases with experience. 
eg: snipers could gradually gain range but not other things, tanks could gain health.

This does allow unit specialization but it could also lead to a quick power tip to the player who is doing the most combat even if they are just spamming defenses with cheap units. This can make the game only approachable with a rush strategy as you try to best your opponent at getting xp. Generally I like to play the turtle with slow growth and impenetrable defenses but this mechanic if not implemented well could make it hard to do a strategy like that. 

How long do you think games should last?
